automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala

```diff
@@ -10,35 +10,48 @@ import org.broadinstitute.dsde.workbench.model.google.GoogleProject
 import org.broadinstitute.dsde.workbench.service.{BillingProject, Orchestration}
 import org.scalatest._
 
-trait GPAllocFixtureSpec extends fixture.FreeSpecLike {
+trait GPAllocFixtureSpec extends fixture.FreeSpecLike with Retries {
   override type FixtureParam = GoogleProject
-  override def withFixture(test: OneArgTest): Outcome =
+  override def withFixture(test: OneArgTest): Outcome = {
     sys.props.get(gpallocProjectKey) match {
       case None                 => throw new RuntimeException("leonardo.billingProject system property is not set")
-      case Some(billingProject) => withFixture(test.toNoArgTest(GoogleProject(billingProject)))
+      case Some(billingProject) =>
+        if (isRetryable(test))
+          withRetry { super.withFixture(test.toNoArgTest(GoogleProject(billingProject))) }
```

Comment:
If a test is retryable but fails all retries, will we end up setting the `shouldUnclaimProjects` property?

Reply:
I'm assuming test will still fail and we'll not unclaim..I'll double check

Reply:
I did a bit more validation...my assumption was wrong...added the same code in the retry branch as well